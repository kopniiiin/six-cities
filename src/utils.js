const MONTHS = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`
];

export const extend = (target, update) => Object.assign({}, target, update);

export const pushElement = (array, element) => [...array, element];

export const replaceElement = (array, element, index) => [...array.slice(0, index), element, ...array.slice(index + 1)];

export const removeElement = (array, index) => [...array.slice(0, index), ...array.slice(index + 1)];

export const upperCaseFirstLetter = (string) => string[0].toUpperCase() + string.slice(1);

export const formatDate = (dateString) => {
  const date = new Date(dateString);
  return `${MONTHS[date.getMonth()]} ${date.getDate()}, ${date.getFullYear()}`;
};
