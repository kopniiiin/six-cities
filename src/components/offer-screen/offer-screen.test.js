import React from "react";
import {shallow} from "enzyme";

import {extend} from "../../utils.js";

import OfferScreen from "./offer-screen.jsx";

import testMocks from "../../test-mocks/offer-screen.js";

describe.skip(`snapshot test: OfferScreen component`, () => {
  it(`should render correctly`, () => expect(
      shallow(<OfferScreen {...testMocks}/>)
  ).toMatchSnapshot());

  it(`should render premium mark`, () => expect(
      shallow(<OfferScreen {...extend(testMocks, {isPremium: true})}/>)
  ).toMatchSnapshot());
});
