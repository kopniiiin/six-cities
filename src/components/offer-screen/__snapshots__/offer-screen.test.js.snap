// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: OfferScreen component should render correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="main.html"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
            key="img/room.jpg"
          >
            <img
              alt="Place photo"
              className="property__image"
              src="img/room.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Paper place
            </h1>
            <BookmarkButton
              blockClassName="property"
              isActive={false}
              isBig={true}
            />
          </div>
          <StarRating
            blockClassName="property"
            isValueShown={true}
            value={4}
          />
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Room
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              4
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              4
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              4
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
                key="paper"
              >
                paper
              </li>
              <li
                className="property__inside-item"
                key="coffee machine"
              >
                coffee machine
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper  user__avatar-wrapper"
              >
                <img
                  alt="Host photo"
                  className="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-max.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Host
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Lots of paper
              </p>
            </div>
          </div>
          <ReviewList
            reviews={
              Array [
                Object {
                  "date": "2020-04-04",
                  "id": "4",
                  "rating": 4,
                  "text": "Good",
                  "user": Object {
                    "name": "User",
                    "photo": Object {
                      "alt": "User photo",
                      "src": "img/avatar-max.jpg",
                    },
                  },
                },
              ]
            }
          />
        </div>
      </div>
      <WithMarkers
        activeMarkerCoordinates={
          Array [
            Array [
              4,
              4,
            ],
          ]
        }
        blockClassName="property"
        markerCoordinates={
          Array [
            Array [
              4,
              4,
            ],
          ]
        }
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <OfferList
          blockClassName="near-places"
          offers={
            Array [
              Object {
                "coordinates": Array [
                  4,
                  4,
                ],
                "id": "4",
                "isFavorite": false,
                "isPremium": false,
                "name": "Paper place",
                "photos": Array [
                  Object {
                    "alt": "Place photo",
                    "src": "img/room.jpg",
                  },
                ],
                "price": 4,
                "rating": 4,
                "type": "room",
              },
            ]
          }
          onOfferCardMouseEnter={[Function]}
          onOfferCardMouseLeave={[Function]}
          onOfferCardNameClick={[Function]}
        />
      </section>
    </div>
  </main>
</div>
`;

exports[`snapshot test: OfferScreen component should render premium mark 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="main.html"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
            key="img/room.jpg"
          >
            <img
              alt="Place photo"
              className="property__image"
              src="img/room.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <PremiumMark
            blockClassName="property"
          />
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Paper place
            </h1>
            <BookmarkButton
              blockClassName="property"
              isActive={false}
              isBig={true}
            />
          </div>
          <StarRating
            blockClassName="property"
            isValueShown={true}
            value={4}
          />
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Room
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              4
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              4
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              4
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
                key="paper"
              >
                paper
              </li>
              <li
                className="property__inside-item"
                key="coffee machine"
              >
                coffee machine
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper  user__avatar-wrapper"
              >
                <img
                  alt="Host photo"
                  className="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-max.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Host
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Lots of paper
              </p>
            </div>
          </div>
          <ReviewList
            reviews={
              Array [
                Object {
                  "date": "2020-04-04",
                  "id": "4",
                  "rating": 4,
                  "text": "Good",
                  "user": Object {
                    "name": "User",
                    "photo": Object {
                      "alt": "User photo",
                      "src": "img/avatar-max.jpg",
                    },
                  },
                },
              ]
            }
          />
        </div>
      </div>
      <WithMarkers
        activeMarkerCoordinates={
          Array [
            Array [
              4,
              4,
            ],
          ]
        }
        blockClassName="property"
        markerCoordinates={
          Array [
            Array [
              4,
              4,
            ],
          ]
        }
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <OfferList
          blockClassName="near-places"
          offers={
            Array [
              Object {
                "coordinates": Array [
                  4,
                  4,
                ],
                "id": "4",
                "isFavorite": false,
                "isPremium": false,
                "name": "Paper place",
                "photos": Array [
                  Object {
                    "alt": "Place photo",
                    "src": "img/room.jpg",
                  },
                ],
                "price": 4,
                "rating": 4,
                "type": "room",
              },
            ]
          }
          onOfferCardMouseEnter={[Function]}
          onOfferCardMouseLeave={[Function]}
          onOfferCardNameClick={[Function]}
        />
      </section>
    </div>
  </main>
</div>
`;
